File: BE/M2/BW2/EnergyService/src/main/java/com/BEBW2/ES/EnergyService/Entities/Provincia.java
package com.BEBW2.ES.EnergyService.Entities;

import lombok.*;

import javax.persistence.*;

@NoArgsConstructor
@Getter
@Setter
@ToString
@Entity
@Table(name = "provincie")
public class Provincia {

    @Setter(AccessLevel.NONE)
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    private Long id;

    private String sigla;
    private String provincia;
    private String regione;

    public Provincia(String sigla, String provincia, String regione) {
        this.sigla = sigla;
        this.provincia = provincia;
        this.regione = regione;
    }
}

File: BE/M2/BW2/EnergyService FE/src/app/models/provincia.ts
export class Provincia {
  id!:number;
  sigla:string;
  provincia:string;
  regione:string;

  constructor(sigla:string, provincia:string, regione:string){
    this.sigla = sigla;
    this.provincia = provincia;
    this.regione = regione;
  }
}
